### ISSUE TITLE
Patch for google maps always opening as directions

### ISSUE THREAD

**Issue by theneekz** [flexible-agency/react-native-map-link#200]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `react-native-map-link@2.8.0` for the project I'm working on.

This is to solve the issue of google maps only opening links as directions rather than single markers when you exclude a sourceLatitude & sourceLongitude.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/react-native-map-link/src/index.js b/node_modules/react-native-map-link/src/index.js
index dbac13d..34c8b9f 100644
--- a/node_modules/react-native-map-link/src/index.js
+++ b/node_modules/react-native-map-link/src/index.js
@@ -129,21 +129,33 @@ export async function showLocation(options) {
     case 'google-maps':
       const googleDirectionMode = getDirectionsModeGoogleMaps();
       // Always using universal URL instead of URI scheme since the latter doesn't support all parameters (#155)
-      url = 'https://www.google.com/maps/dir/?api=1';
       if (useSourceDestiny) {
+        url = 'https://www.google.com/maps/dir/?api=1';
         url += `&origin=${sourceLatLng}`;
-      }
-      if (!options.googleForceLatLon && title) {
-        url += `&destination=${encodedTitle}`;
+        if (!options.googleForceLatLon && title) {
+          url += `&destination=${encodedTitle}`;
+        } else {
+          url += `&destination=${latlng}`;
+        }
+
+        url += options.googlePlaceId
+          ? `&destination_place_id=${options.googlePlaceId}`
+          : '';
+
+        url += googleDirectionMode ? `&travelmode=${googleDirectionMode}` : '';
       } else {
-        url += `&destination=${latlng}`;
-      }
+        url = 'https://www.google.com/maps/search/?api=1';
 
-      url += options.googlePlaceId
-        ? `&destination_place_id=${options.googlePlaceId}`
-        : '';
+        if (!options.googleForceLatLon && title) {
+          url += `&query=${encodedTitle}`;
+        } else {
+          url += `&query=${latlng}`;
+        }
 
-      url += googleDirectionMode ? `&travelmode=${googleDirectionMode}` : '';
+        url += options.googlePlaceId
+        ? `&query_place_id=${options.googlePlaceId}`
+        : '';
+      }
       break;
     case 'citymapper':
       url = `${prefixes.citymapper}directions?endcoord=${latlng}`;
@@ -155,6 +167,7 @@ export async function showLocation(options) {
       if (useSourceDestiny) {
         url += `&startcoord=${sourceLatLng}`;
       }
+      console.log({url})
       break;
     case 'uber':
       url = `${prefixes.uber}?action=setPickup&dropoff[latitude]=${lat}&dropoff[longitude]=${lng}`;
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by theneekz** [comment]
[PR link](https://github.com/flexible-agency/react-native-map-link/pull/201/files)

**Comment by tschoffelen** [comment]
Fixed in #201
